src/mesh/mod.rs

```diff
@@ -114,23 +110,53 @@ impl Mesh {
                                 &vertices_collection[cast_usize(v3)],
                             );
 
-                            normals_collection.push(face_normal);
-
+                            normals_collection_sharp.push(face_normal);
                         }
-                        // FIXME: once we add other kinds of faces, they must panic here
-                        // _ => panic!("Face must be a triangle, we just created it"),
                     }
                 }
+                assert_eq!(normals_collection_sharp.len(), faces_collection_sharp.len());
+                assert_eq!(
+                    normals_collection_sharp.capacity(),
+                    faces_collection_sharp.len()
+                );
```

Comment:
:heart:

Reply:
?

Reply:
Really? I thought the assert wasn't present previously, but it indeed is good that we now have it. Or maybe I just didn't see it.